core/fetcher.go

```diff
@@ -160,64 +171,82 @@ func (f *BlockFetcher) ValidatorSet(ctx context.Context, height int64) (*types.V
 	return validatorSet, nil
 }
 
-// SubscribeNewBlockEvent subscribes to new block events from Core, returning
-// a new block event channel on success.
-func (f *BlockFetcher) SubscribeNewBlockEvent(ctx context.Context) (<-chan types.EventDataSignedBlock, error) {
-	if f.isListeningForBlocks.Load() {
-		return nil, fmt.Errorf("already subscribed to new blocks")
-	}
-	ctx, cancel := context.WithCancel(ctx)
-	f.cancel = cancel
-	f.doneCh = make(chan struct{})
-	f.isListeningForBlocks.Store(true)
+func (f *BlockFetcher) runSubscriber() (chan types.EventDataSignedBlock, error) {
+	var (
+		subscription coregrpc.BlockAPI_SubscribeNewHeightsClient
+		err          error
+	)
```

Comment:
subscription and error do not need to be predeclared here, since it only propagated down the stack from for loop